src/viking.js

```diff
@@ -1,11 +1,116 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+  this.health = health;
+  this.strength = strength;
+}
+
+Soldier.prototype.attack = function() {
+  return this.strength;
+};
+
+Soldier.prototype.receiveDamage = function(damage) {
+  this.health -= damage;
+};
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+  this.name = name;
+  Soldier.call(this, health, strength);
+}
+
+Viking.prototype = new Soldier();
+
+Viking.prototype.receiveDamage = function(damage) {
+  this.health -= damage;
+  if (this.health > 0) {
+    return this.name + " has received " + damage + " points of damage";
+  } else {
+    return this.name + " has died in act of combat";
+  }
+};
+
+Viking.prototype.battleCry = function() {
+  return "Odin Owns You All!";
+};
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+  Soldier.call(this, health, strength);
+}
+
+Saxon.prototype = new Soldier();
+
+Saxon.prototype.attack = function() {
+  return this.strength;
+};
+
+Saxon.prototype.receiveDamage = function(damage) {
+  this.health -= damage;
+  if (this.health > 0) {
+    return "A Saxon has received " + damage + " points of damage";
+  } else {
+    return "A Saxon has died in combat";
+  }
+};
 
 // War
-function War() {}
+function War() {
+  this.vikingArmy = [];
+  this.saxonArmy = [];
+}
+
+War.prototype.addViking = function(viking) {
+  this.vikingArmy.push(viking);
+};
+
+War.prototype.addSaxon = function(saxon) {
+  this.saxonArmy.push(saxon);
+};
+
+War.prototype.vikingAttack = function() {
+  var randomNumberForViking = Math.floor(
+    Math.random() * this.vikingArmy.length
+  );
+  var randomNumberForSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+
+  var randomSaxon = this.saxonArmy[randomNumberForSaxon];
+  var randomViking = this.vikingArmy[randomNumberForViking];
+
+  // var damage = randomViking.attack();
```

Comment:
Try not to leave code commented, if other developer work on this project they don't know if this is not used anymore or that it will be used or you just were trying things